linera-storage/src/lib.rs

```diff
@@ -321,28 +339,49 @@ pub trait Storage: Sized {
         let compressed_service_bytecode = CompressedBytecode {
             compressed_bytes: service_blob.into_bytes().to_vec(),
         };
-        let service_bytecode = linera_base::task::Blocking::<linera_base::task::NoInput, _>::spawn(
-            move |_| async move { compressed_service_bytecode.decompress() },
-        )
-        .await
-        .join()
-        .await?;
-        Ok(WasmServiceModule::new(service_bytecode, wasm_runtime)
-            .await?
-            .into())
-    }
-
-    #[cfg(not(with_wasm_runtime))]
-    #[allow(clippy::diverging_sub_expression)]
-    async fn load_service(
-        &self,
-        _application_description: &UserApplicationDescription,
-    ) -> Result<UserServiceCode, ExecutionError> {
-        panic!(
-            "A Wasm runtime is required to load user applications. \
-            Please enable the `wasmer` or the `wasmtime` feature flags \
-            when compiling `linera-storage`."
-        );
+        let _service_bytecode =
+            linera_base::task::Blocking::<linera_base::task::NoInput, _>::spawn(
+                move |_| async move { compressed_service_bytecode.decompress() },
+            )
+            .await
+            .join()
+            .await?;
+        match application_description.bytecode_id.vm_runtime {
+            VmRuntime::Wasm => {
+                cfg_if::cfg_if! {
+                    if #[cfg(with_wasm_runtime)] {
+                        let Some(wasm_runtime) = self.wasm_runtime() else {
+                            panic!("A Wasm runtime is required to load user applications.");
+                        };
+                        Ok(WasmServiceModule::new(_service_bytecode, wasm_runtime)
+                           .await?
+                           .into())
+                    } else {
+                        panic!(
+                            "A Wasm runtime is required to load user applications. \
+                             Please enable the `wasmer` or the `wasmtime` feature flags \
+                             when compiling `linera-storage`."
+                        );
+                    }
+                }
+            }
+            VmRuntime::Evm => {
+                cfg_if::cfg_if! {
+                    if #[cfg(with_revm)] {
+                        let evm_runtime = EvmRuntime::Revm;
+                        Ok(EvmServiceModule::new(_service_bytecode, evm_runtime)
+                           .await?
+                           .into())
```

Comment:
I'm not really sure what it means to have an Evm service. We could either return an error saying Evm services are not supported, or figure out a way to query the Evm state, either exposing some read-only entrypoints or a single generic service to inspect the state.

Reply:
What it means to have service means to call the smart contract with the contract being executed without the result being written to the blockchain.

The corresponding function in the Ethereum blockchain is `eth_call`.